### PR TITLE
[jit] Use r8 for r4 stack vars to avoid problems with stack merges when one side is an r4 and the other an r8. Add conversions for compares.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/method-to-ir.c [hunk 1]
```diff
@@ -1436,10 +1436,20 @@ mono_compile_get_interface_var (MonoCompile *cfg, int slot, MonoInst *ins)
...
 	MonoInst *res;
 	int pos, vnum;
+	MonoType *type;
+
+	/*
+	 * Use double for r4 stack vars to avoid narrowing with r4-r8 stack
+	 * merges.
+	 */
+	if (ins->type == STACK_R4)
+		type = &mono_defaults.double_class->byval_arg;
```
Comment (luhenry) [inline mono/mini/method-to-ir.c]: This will require a `m_class_get_byval_arg`.

**Comment by vargaz** [comment]
Thats kinda hard to do since by the time we realize we need an r8, the variable is already allocated and assigned. Note that this is the same behavior as the one without -O=float32.

**Comment by kumpera** [comment]
We can backpatch the r4 site to convert & store to a r8.

**Comment by vargaz** [comment]
Will add that as a fixme.

**Comment by luhenry** [comment]
@monojenkins build Linux x64

**Comment by luhenry** [comment]
@monojenkins build Linux i386

**Pull request merged**